rs/replicated_state/src/page_map/storage.rs

```diff
@@ -884,6 +886,70 @@ impl dyn StorageLayout + '_ {
         }
         Ok(result)
     }
+
+    // Read the number of memory pages from overlay.
+    // Basically it's the index of the last page, which we read based on the offset from the end of
+    // the file plus some error handling.
+    fn overlay_memory_pages(overlay: &Path) -> StorageResult<usize> {
```

Comment:
The naming is a bit ambiguous (number of non-zero pages vs number of logical pages), but I can't really think of a strictly better name. We called it `num_logical_pages` somewhere else.